docs/design/generics/details.md

```diff
@@ -2614,6 +2606,9 @@ fn Contains
     (haystack: SC, needles: CT) -> bool;
 ```
 
+A constraint setting an associated type to a concrete type, like
+`where .ElementType == String`, gives `.ElementType` the complete `String` API.
```

Comment:
I wonder if something like this would be helpful to clarify:

```suggestion
`where .ElementType == String`, gives `.ElementType` the complete `String` API,
in addition to any API from the type of `ElementType`.
```